Detectors/TPC/calibration/include/TPCCalibration/CalibPadGainTracks.h

```diff
@@ -0,0 +1,233 @@
+  {
+    vTracks = vTPCTracksArrayInp;
+  }
+
+  /// set the momentum range which will be used
+  void setMomentumRange(float momMin, float momMax)
+  {
+    mMomMin = momMin;
+    mMomMax = momMax;
+  }
+
+  /// this function sets the mode of the class.
+  /// e.g. mode=0 -> use the truncated mean from the track for normalizing the dedx
+  ///      mode=1 -> use the value from the BB-fit for normalizing the dedx
+  void setMode(int iMode)
+  {
+    mMode = iMode;
+  }
+
+  /// processes the member variables set by 'setClnative()'  and 'setTracks()' and calls the processTrack function
+  void processTracks(bool writeTree);
+
+  /// write the relevant variables (e.g.: sector,padRow,padinRow,qMax,dEdx ) used by this class to file
+  void writeTree();
+
+  /// perform truncation of qMax in each pad and fill the obtained gain in CalPad object
+  void makeTruncPad();
+
+  /// this functions compares the extracted map with some reference gainmap
+  /// \param folder output folder for plots
+  /// \param refMap name of reference gainmap
+  /// \param outFileType file type of output
+  void compareMaps(string folder = "", std::string refMap = "RefGainMap.root", std::string outFileType = "pdf");
+
+  /// this functions plots the reference map
+  /// \param folder output folder for plots
+  /// \param outFileType file type of output
+  void plotExtractedGainmap(string folder = "", std::string outFileType = "pdf");
+
+  /// make logarithmic
+  /// \param nbinsX number of bins
+  /// \param xmin minimum x value
+  /// \param xmax maximum x value
+  static std::vector<double> makeLogBinning(int nbinsX, float xmin, float xmax);
+
+ private:
+  /// calculate truncated mean for track
+  void processTrack(o2::tpc::TrackTPC& track);
+
+  /// set which bethe bloch parametrization will be used for fitting
+  /// \param iBBModel selecte the BB model 0=ALEPH, 1=MODIFIED_ALEPH, 2=LUND, 3=SATURATED_LUND
+  void setBBFunction(int iBBModel);
+
+  /// initialize the histograms
+  void initializeHisto();
+
+  /// make linear
+  /// \param nbinsX number of bins
+  /// \param xmin minimum x value
+  /// \param xmax maximum x value
+  std::vector<double> makeLinBinning(int nbinsX, float xmin, float xmax);
+
+  ///get the truncated mean for input vector and the truncation range low*nCl<nCl<high*nCl
+  /// \param vCharge vector containing all qmax values of the track
+  /// \param low lower cluster cut of  0.05*nCluster
+  /// \param high higher cluster cut of  0.6*nCluster
+  float getTruncMean(vector<float> vCharge, float low = 0.05f, float high = 0.6f);
+
+  /// \return dedx in IROC/OROC/ALL for input track
+  std::array<float, 3> getdEdxfromTrack(o2::tpc::TrackTPC& track);
+
+  /// get the filled histograms
+  std::array<TH2F, 2> getHisto();
+
+  /// fill the extracted gainvalues in CalPad object
+  /// \param arrgain array containing for each pad in each sector(36) the gain value
+  void fillgainMap(std::array<std::array<float, 14560>, 36>& arrgain);
```

Comment:
Do I understand correctly, that this class is needed for producing the calibration, but not for applying it?

Application of the calibration might eventually happen on the GPU, which will not support std::array, so in that case please use normal arrays.

Reply:
> Do I understand correctly, that this class is needed for producing the calibration, but not for applying it?
> 
> Application of the calibration might eventually happen on the GPU, which will not support std::array, so in that case please use normal arrays.

Hi @davidrohr, yes this class only produces the pad-by-pad gainmap. The application of the gainmap will not be performed in this class.